### ISSUE TITLE
HCL Parsing doesn't report aws_db_instance's cost when it's a replica

### ISSUE THREAD

**Issue by kahnwong** [infracost/infracost#1436]
## output from`infracost breakdown --path .`
```
 Name                                                              Monthly Qty  Unit               Monthly Cost (THB)

 module.dmp.aws_db_instance.db
 ├─ Database instance (on-demand, Single-AZ, db.t3.xlarge)                 730  hours                      ฿10,824.97
 ├─ Storage (general purpose SSD, gp2)                                     100  GB                            ฿456.78
 ├─ Additional backup storage                                Monthly cost depends on usage: ฿3.14 per GB
 ├─ Performance Insights Long Term Retention (db.t3.xlarge)                  4  vCPU-month                    ฿457.57
 └─ Performance Insights API                                 Monthly cost depends on usage: ฿0.33 per 1000 requests

 module.dmp.aws_route53_record.db
 ├─ Standard queries (first 1B)                              Monthly cost depends on usage: ฿13.24 per 1M queries
 ├─ Latency based routing queries (first 1B)                 Monthly cost depends on usage: ฿19.86 per 1M queries
 └─ Geo DNS queries (first 1B)                               Monthly cost depends on usage: ฿23.17 per 1M queries

 module.dmp_replica.aws_db_instance.db
 ├─ Database instance (on-demand, Single-AZ, db.t3.xlarge)                 730  hours                      ฿10,824.97
 ├─ Storage (general purpose SSD, gp2)                                     100  GB                            ฿456.78
 └─ Performance Insights API                                 Monthly cost depends on usage: ฿0.33 per 1000 requests

 module.dmp_replica.aws_route53_record.db
 ├─ Standard queries (first 1B)                              Monthly cost depends on usage: ฿13.24 per 1M queries
 ├─ Latency based routing queries (first 1B)                 Monthly cost depends on usage: ฿19.86 per 1M queries
 └─ Geo DNS queries (first 1B)                               Monthly cost depends on usage: ฿23.17 per 1M queries

 OVERALL TOTAL (THB)                                                                                       ฿23,021.07
──────────────────────────────────
6 cloud resources were detected, rerun with --show-skipped to see details:
∙ 4 were estimated, 4 include usage-based costs, see https://infracost.io/usage-file
∙ 2 were free

Add cost estimates to your pull requests: https://infracost.io/cicd
```

## output from `infracost breakdown --path=./ --terraform-parse-hcl`
`module.dmp_replica.aws_db_instance.db` is reported as `฿0`

```
 Name                                                             Monthly Qty  Unit               Monthly Cost (THB)

 module.dmp.aws_db_instance.db
 ├─ Database instance (on-demand, Single-AZ, db.t3.xlarge)                730  hours                      ฿10,824.97
 ├─ Storage (general purpose SSD, gp2)                                    100  GB                            ฿456.78
 ├─ Additional backup storage                               Monthly cost depends on usage: ฿3.14 per GB
 └─ Performance Insights API                                Monthly cost depends on usage: ฿0.33 per 1000 requests

 module.dmp.aws_route53_record.db
 ├─ Standard queries (first 1B)                             Monthly cost depends on usage: ฿13.24 per 1M queries
 ├─ Latency based routing queries (first 1B)                Monthly cost depends on usage: ฿19.86 per 1M queries
 └─ Geo DNS queries (first 1B)                              Monthly cost depends on usage: ฿23.17 per 1M queries

 module.dmp_replica.aws_db_instance.db
 ├─ Database instance (on-demand, Single-AZ, db.t3.xlarge)                730  hours                           ฿0.00
 ├─ Storage (general purpose SSD, gp2)                                    100  GB                            ฿456.78
 └─ Performance Insights API                                Monthly cost depends on usage: ฿0.33 per 1000 requests

 module.dmp_replica.aws_route53_record.db
 ├─ Standard queries (first 1B)                             Monthly cost depends on usage: ฿13.24 per 1M queries
 ├─ Latency based routing queries (first 1B)                Monthly cost depends on usage: ฿19.86 per 1M queries
 └─ Geo DNS queries (first 1B)                              Monthly cost depends on usage: ฿23.17 per 1M queries

 OVERALL TOTAL (THB)                                                                                      ฿11,738.53
──────────────────────────────────
6 cloud resources were detected, rerun with --show-skipped to see details:
∙ 4 were estimated, 4 include usage-based costs, see https://infracost.io/usage-file
∙ 2 were free

Add cost estimates to your pull requests: https://infracost.io/cicd
```

on a slightly-related note, our CTO and CFO love the html report!

**Comment by vdmgolub** [comment]
@kahnwong Hello! Thank you for creating the issue! Will it be possible to share a sanitized TF code with your configuration so we can reproduce it? 

We're happy to see that it is useful for your org! 😍 May I ask if you mean by the "html report" our Dashboard [share links](https://www.infracost.io/docs/features/share_links/) feature or the HTML [output format](https://www.infracost.io/docs/features/cli_commands/#combined-output-formats)?

**Comment by kahnwong** [comment]
Whoa dashboard report is now a thing! Gotta try that out stat.

By HTML report I mean `infracost.html` generated from `breakdown_all.sh`

also terraform blocks:
### module/main.tf
```hcl
#tfsec:ignore:aws-rds-encrypt-instance-storage-data
#tfsec:ignore:aws-rds-specify-backup-retention
resource "aws_db_instance" "db" {
  engine            = var.engine
  allocated_storage = var.allocated_storage
  engine_version    = var.engine_version
  instance_class    = var.instance_class

  identifier = var.identifier
  db_name    = var.db_name
  username   = var.username
  password   = var.password

  replicate_source_db = var.source_db_id

  parameter_group_name = var.parameter_group_name

  performance_insights_enabled    = true
  performance_insights_kms_key_id = local.performance_insights_kms_key_id

  deletion_protection = var.deletion_protection

  ### networking
  vpc_security_group_ids = local.vpc_security_group_ids
  db_subnet_group_name   = var.db_subnet_group_name == true ? local.rds_subnet : null

  ### backup (in UTC)
  maintenance_window      = var.maintenance_window
  backup_window           = var.backup_window
  backup_retention_period = var.backup_window != null ? 3 : 0 # Backups are required in order to create a replica
  skip_final_snapshot     = var.skip_final_snapshot

  #   apply_immediately = true # adhoc only
  #   allow_major_version_upgrade = false

}
```

### module/variables.tf
```hcl
variable "allocated_storage" {
  type = number
}
variable "engine_version" {
  type    = string
  default = null
}
variable "db_name" {
  type    = string
  default = null
}
variable "parameter_group_name" {
  type    = string
  default = null
}
variable "instance_class" {
  type = string
}
variable "identifier" {
  type = string
}
variable "username" {
  type    = string
  default = null
}
variable "password" {
  type    = string
  default = null
}
variable "deletion_protection" {
  type = bool
}
variable "skip_final_snapshot" {
  type = bool
}
variable "dns_name" {
  type = string
}
variable "source_db_id" {
  type    = string
  default = null
}
variable "engine" {
  type    = string
  default = "postgres"
}
variable "db_subnet_group_name" {
  type    = string
  default = false
}
variable "maintenance_window" {
  type    = string # 19:00-21:00 in Asia/Bangkok
  default = "Sun:12:00-Sun:14:00"
}
variable "backup_window" {
  type    = string
  default = "14:00-16:00" # 21:00-23:00 in Asia/Bangkok
}
```

### rds.tf
```hcl
module "db" {
  source = "../modules/rds"

  engine_version    = var.engine_version
  allocated_storage = var.allocated_storage
  instance_class    = var.instance_class

  identifier = "#######"
  db_name    = "#######"

  username = "#######"
  password = "#######"

  parameter_group_name = aws_db_parameter_group.this.name
  deletion_protection  = true
  skip_final_snapshot  = true
}

module "db_replica" {
  source = "../modules/rds"

  engine = null # disable for replica

  allocated_storage = var.allocated_storage
  instance_class    = var.instance_class

  identifier = "#######"

  db_subnet_group_name = false # disable for replica
  parameter_group_name = aws_db_parameter_group.this.name
  deletion_protection  = true
  backup_window        = null # disable for replica
  skip_final_snapshot  = true

  source_db_id = module.db.db_id
}
```

**Comment by vdmgolub** [comment]
Awesome, thank you! It seems like the issue is that replica's `engine` is not set. We'll look how to get it set using the source DB.

Yeah, when you enable the Dashboard support, each run will generate a new unique link that can be easily shared. Currently these links do not expire.

I also noticed that you're using our bash script for Terragrunt. Lately we worked on optimizing our CLI dev experience and it should be easier now to use `infracost breakdown` and `infracost comment` commands directly. Here's an example how to use them with GitHub Actions: https://github.com/infracost/actions/tree/master/examples/terragrunt

**Comment by vdmgolub** [comment]
@kahnwong We've just merged a fix for the `aws_db_instance` resource. It takes care of retrieving an engine for replicas. It will go live with the next release. We'll notify here when it's out.

We've also discovered that TF plan parser skips a reference if its value is provided by a variable. I created the issue #1442 to track and fix that. I've closed this issue for now (well, GitHub automation did after the PR merge :) ), thank you for creating it!
